File: app/src/main/java/com/coinninja/coinkeeper/ui/actionbar/managers/TitleViewManager.java
package com.coinninja.coinkeeper.ui.actionbar.managers;


import android.view.View;
import android.widget.TextView;

import javax.inject.Inject;

import androidx.appcompat.app.ActionBar;

public class TitleViewManager {
    private ActionBar actionBar;
    private TextView titleView;

    @Inject
    public TitleViewManager() {
    }

    public void renderTitleView() {
        String title = getTitle();
        if (isTitleValid(title)) {
            setTitle(title);
        } else {
            removeTitle();
        }
    }

    public void renderTitleView(String string) {
        setTitle(string);
    }

    public void renderUpperCaseTitleView() {
        String title = getTitle();
        if (isTitleValid(title)) {
            setTitle(title.toUpperCase());
        } else {
            removeTitle();
        }

    }

    public String getTitle() {
        String title = actionBar.getTitle().toString();
        actionBar.setTitle("");
        return title;
    }


    private void setTitle(String title) {
        if ((title != null && title.equals("")) || title == null) {
            titleView.setVisibility(View.GONE);
            return;
        }

        titleView.setVisibility(View.VISIBLE);
        titleView.setText(title);
    }

    private void removeTitle() {
        titleView.setVisibility(View.GONE);
    }

    public void setActionBar(ActionBar actionBar) {
        this.actionBar = actionBar;
    }

    public void setTitleView(TextView titleView) {
        this.titleView = titleView;
    }

    public boolean isTitleValid(String title) {
        return title != null && !title.isEmpty();
    }
}

File: app/src/main/java/com/coinninja/coinkeeper/ui/actionbar/managers/TitleViewManager.kt
package com.coinninja.coinkeeper.ui.actionbar.managers


import android.view.View
import android.widget.TextView
import androidx.appcompat.app.ActionBar
import app.dropbit.annotations.Mockable
import javax.inject.Inject

@Mockable
class TitleViewManager @Inject constructor() {
    //TODO make this A Constructor argument
    var actionBar: ActionBar? = null

    //TODO make this A Constructor argument
    var titleView: TextView? = null

    private var _title = ""

    var title: String
        get() {
            return if (_title.isNotEmpty()) {
                _title
            } else {
                _title = actionBar?.title?.toString() ?: ""
                actionBar?.title = ""
                _title
            }
        }
        set(value) {
            actionBar?.title = ""
            _title = value
        }

    fun renderTitle() {
        if (title.isNotEmpty()) {
            titleView?.visibility = View.VISIBLE
            titleView?.text = title.toUpperCase()
        } else {
            removeTitle()
        }
    }

    private fun removeTitle() {
        titleView?.visibility = View.GONE
    }
}
